source/states/BaseState.hx

```diff
@@ -0,0 +1,96 @@
+package states;
+
+import flixel.FlxObject;
+
+/**
+ * General state class for representing states for any generic FlxObject
+ *
+ * As a note, a state should be responsible for calling its nextState's enter
+ * method, and also responsible for calling its own exit method
+ *
+ * @author Garren Ijames
+ */
+class BaseState
+{
+  public  var nextState:Null<BaseState>;
+
+  private var duration:Int;
```

Comment:
Since we're using acceleration and velocity in this game, I think we might want to move away from measuring the duration of a state in frames. Or maybe not. I'm pretty torn, because I think animations in HaxeFlixel are still done on a frame-by-frame basis.

Do you think it would be too confusing to have two duration variables, one that's an int measuring frames, and the other that's a float measuring time (in seconds)?

You could use a similar system where -1 signals that the duration isn't measured with that variable, and otherwise if either runs out, the nextState is activated.